test/src/e2e_vm_tests/mod.rs

```diff
@@ -462,7 +462,12 @@ pub fn run(filter_regex: Option<regex::Regex>) {
         contract_ids.push(contract_id);
     }
     for name in projects {
-        harness::runs_on_node(name, &contract_ids);
+        assert!(!harness::runs_on_node(name, &contract_ids)
+            .iter()
+            .any(|r| matches!(
+                r,
+                fuel_tx::Receipt::Revert { .. } | fuel_tx::Receipt::Panic { .. }
```

Comment:
Two things:

1. I prefer `all` instead of `!any` just due to personal cognitive overhead of logical negation.
2. Do we want to allow configurable expected return receipts instead of just checking that it didn't panic here? Allow the harness to expect a certain response from the contract call, like how we do with other script tests?

Reply:
We could probably check the return from main (so check the last return?). Checking that every contract call is correct is, I think, too much. I feel it's better to encourage asserts in the script instead as this would make modifying the test a lot simpler.

Reply:
So, I modified this a big to also allow checking for the last `return` (i.e. second to last entry in the vector of receipts). I don't like this very much as I prefer to get access to `ProgramState` but I can't do that from a vector of receipts. Matching on receipts directly is also a big annoying because each variant requires specifying many parameters. So, I require matching on returned integers directly.